Fixes #37951 - Appropriate generic default value for the UI

pavanshekar opened the pull request:
#### What are the changes introduced in this pull request?

Updated the code to use the default value of the generic content type in the new repository creation form.

#### Considerations taken when implementing this change?

This change is not limited to the depth field. All generic repository options will now be set to its respective default value.

#### What are the testing steps for this pull request?

1. Go to Products -> Any Product ->New Repository
2. Select OSTree as Content Type and fill only the required fields
3. Notice that the default value of the Depth field is set to 0
4. Able to save and create a New Repository

ianballou commented on engines/bastion_katello/app/assets/javascripts/bastion_katello/products/details/repositories/new/new-repository.controller.js:
```diff
@@ -96,7 +96,7 @@ angular.module('Bastion.repositories').controller('NewRepositoryController',
                 if ($scope.genericRemoteOptions && $scope.genericRemoteOptions !== []) {
                     $scope.genericRemoteOptions.forEach(function(option) {
-                        option.value = "";
+                        option.value = option.default || "";
```
When `option.default` resolves to 0, JS considers it "falsey" and thus chooses empty string `""`.  In my testing, the Pulp remote gets sent empty string as the depth value for a new ostree repo with no depth explicitly set in the UI.

ianballou reviewed (changes requested):
Saw one issue, but when I hacked around it default option did pop up in the text field.

ianballou reviewed (commented):
One suggestion -- for yum repositories, there is a Metadata Expiration field. It has a default value of 1, but no value actually needs to be populated in the field. If left empty, the default is used.
Could this be modified so that the integer field could be left empty to use the default? 
Example: 
![image](https://github.com/user-attachments/assets/41196939-cc1a-4c3e-a2fd-04583340da4b)

ianballou reviewed (commented):
One somewhat-related request, do you mind updating the depth field so that it does not have any delimiter? It was added by accident. https://github.com/Katello/katello/blob/master/lib/katello/repository_types/ostree.rb#L27

pavanshekar commented:
Fixed the following:

1. Similar to YUM repositories example shared, for OSTree repositories, the depth field will now default to 0 if left empty. The default value of 0 will be passed to the Pulp remote.
2. Removed the delimiter from the depth field as it was added by mistake. The depth field now functions without any delimiter.

And have pushed the updated code.

ianballou commented:
Don't worry about the EL 8 packit failure -- we need to remove that.

ianballou reviewed (approved):
Works great, thanks!
